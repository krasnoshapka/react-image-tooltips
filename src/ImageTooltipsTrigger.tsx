import * as React from 'react';

export interface ImageTooltipsTriggerProps extends React.ComponentPropsWithoutRef<"div"> {
  className?: string;
  children: React.ReactNode;
  handleClick?: () => void
}

export const ImageTooltipsTrigger: React.FC<ImageTooltipsTriggerProps> = ({
  className, children, handleClick, ...props
}: ImageTooltipsTriggerProps) => {
  const cssClass = ["hotspot-trigger", className];
  return (
    <div className={cssClass.join(' ')} {...props} tabIndex={0} onClick={handleClick}>
      {children}
    </div>
  );
}