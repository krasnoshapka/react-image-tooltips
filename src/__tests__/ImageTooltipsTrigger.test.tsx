import * as React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {ImageTooltipsTrigger} from '../ImageTooltipsTrigger';

describe('<ImageTooltipsTrigger />', () => {
  it('renders correctly', () => {
    const {container} = render(
      <ImageTooltipsTrigger className="test-trigger">+</ImageTooltipsTrigger>
    );

    expect(container).toMatchSnapshot();
  });

  it('appends passed classnames', () => {
    const {container} = render(
      <ImageTooltipsTrigger className="test-trigger">+</ImageTooltipsTrigger>
    );

    expect(container.firstChild).toHaveClass('test-trigger');
  });

  it('forwards extra props', () => {
    const {container} = render(
      <ImageTooltipsTrigger id="test-id">+</ImageTooltipsTrigger>
    );

    expect(container.firstChild).toHaveAttribute('id', 'test-id');
  });

  it('calls handleClick', () => {
    const handleClick = jest.fn();

    const {container} = render(
      <ImageTooltipsTrigger handleClick={handleClick}>+</ImageTooltipsTrigger>
    );

    expect(handleClick).not.toBeCalled();

    fireEvent.click(container.firstChild!);

    expect(handleClick).toBeCalled();
  });

});