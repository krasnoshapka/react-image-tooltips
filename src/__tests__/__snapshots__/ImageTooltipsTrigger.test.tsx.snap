// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageTooltipsTrigger /> renders correctly 1`] = `
<div>
  <div
    class="hotspot-trigger test-trigger"
    tabindex="0"
  >
    +
  </div>
</div>
`;
